### DISCUSSION TITLE
nest-commander not recognizing module imports

### DISCUSSION THREAD

**Discussion by leemaria** [jmcdo29/nest-commander#772] [Q&A] [answered]
Hello, 

I am trying to implement nest-commander for my project, which is currently using Standalone applications

Without nest-commander, the setup is like this:

main.ts
```
async function bootstrap() {
  await CommandFactory.run(AppModule, ['warn', 'error']);

  if (process.argv[2] === 'basic') {
    const app = await NestFactory.createApplicationContext(AppModule);
    const tasksService = app.get(DummyTestService);
    await tasksService.run();
    await app.close();
  }

}

bootstrap();
```

app.module.ts
```
@Module({
  imports: [CustomConfigModule],
  providers: [DummyTestService]
})
export class AppModule {}
```

custom-config.module.ts
```
@Module({
  providers: [
    {
      provide: CustomConfigService,
      useFactory: async () => {
        const customConfig = await CustomConfigService.getInstance();
        return customConfig;
      }
    }
  ],
  exports: [CustomConfigService]
})
export class CustomConfigModule {}
}
```

dummy-test.service.ts
```
@Injectable()
export class DummyTestService {
  private static readonly logger = new Logger(DummyTestService.name);

  async run(): Promise<void> {
    console.log('hello');

    DummyTestService.logger.log('Hello World');
    return;
  }
}
```

My CustomConfigService initializes some authorization stuff and uses the Logger to print to the console when it completes its tasks. When I run my Nest JS app normally with argument "basic", I get all the Logs from my CustomConfigService and DummyTestService. 

When I try to switch to using nest-commander, I am only making these 2 changes to

main.ts
```
async function bootstrap() {
  await CommandFactory.run(AppModule);
}

bootstrap();
```

and dummy-test.service.ts
```
@Command({ name: 'basic', description: 'A parameter parse' })
@Injectable()
export class DummyTestService {
  private static readonly logger = new Logger(DummyTestService.name);

  async run(): Promise<void> {
    console.log('hello');

    DummyTestService.logger.log('Hello World');
    return;
  }
}
```

Running the command now gives me "hello" (the console.log from dummy-test.service.ts, but not any of my Logger.log. It seems like my CustomConfigModule is not getting loaded in properly. 

Does anyone happen to have insight as to what might be happening?

**Comment by jmcdo29** [comment, marked as answer]
Huh, that's weird. Do you have a minimum reproduction for this at all? I'd love to take a look

**Reply by leemaria** [reply]
Yes, here is a codesandbox I put together quickly and got the same behavior

As a standalone application: https://codesandbox.io/s/standalone-application-p9u92l

Using nest-commander: https://codesandbox.io/s/nest-commander-vq3n9q

As mentioned, the only changes between the two are in `main.ts` and `/services/dummy-test.service.ts`

**Reply by jmcdo29** [reply]
Ah, you know what I think it is? `nest-commander` uses `logger: false` under the hood when calling `NestFactory.createApplicationContext()`, to cut down on the start up logs as the CLI should be relatively silent. So when you're using Nest's `Logger`, that is set to `false` and doesn't print anything, all the options are a no-op. You can use `CommandFactory.run(AppModule, { logger: new Logger() })` and it should behave similarly to the direct approach

**Reply by leemaria** [reply]
Oh my, that was exactly it! So it was working this entire time. Thank you so much!
